ludwig/models/trainer.py

```diff
@@ -91,40 +94,84 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self.shutdown()
 
 
+@dataclass
+class TrainerConfig:
+    optimizer: Dict = schema.OptimizerOptions(default={TYPE: "adam"})
+    epochs: int = schema.PositiveInteger(default=100)
+    regularization_lambda: float = schema.FloatRange(default=0.0, min=0)
+    regularization_type: Optional[str] = schema.StringOptions(
+        options=["l1", "l2", "l1_l2"], default="l2", nullable=True
+    )
+    learning_rate: float = schema.NumericOrStringOptionsField(
+        default=0.001, min=0.0, max=1.0, options=["auto"], nullable=False
+    )
+    batch_size: Union[int, str] = schema.IntegerOrStringOptionsField(
+        default=128, options=["auto"], nullable=False, exclusiveMin=0
+    )
+    eval_batch_size: Optional[int] = schema.IntegerOrStringOptionsField(
+        default=None, options=["auto"], nullable=True, exclusiveMin=0
+    )
+    early_stop: int = schema.IntegerRange(default=5, min=-1)  # Can be -1, which disables early_stop
+    reduce_learning_rate_on_plateau: float = schema.FloatRange(default=0.0, min=0.0, max=1.0)
+    reduce_learning_rate_on_plateau_patience: int = schema.NonNegativeInteger(5)
+    reduce_learning_rate_on_plateau_rate: float = schema.FloatRange(default=0.5, min=0.0, max=1.0)
+    increase_batch_size_on_plateau: int = schema.NonNegativeInteger(default=0)
+    increase_batch_size_on_plateau_patience: int = schema.NonNegativeInteger(default=5)
+    increase_batch_size_on_plateau_rate: float = schema.FloatRange(default=2.0, min=0.0)
+    increase_batch_size_on_plateau_max: int = schema.PositiveInteger(default=512)
+    decay: bool = False
+    decay_steps: int = schema.PositiveInteger(default=10000)
+    decay_rate: float = schema.FloatRange(default=0.96, min=0.0, max=1.0)
+    staircase: bool = False
+    # "gradient_clipping": None, # TODO: What is this?
+    # TODO: Need some more logic here for validating against output features
+    validation_field: str = COMBINED
+    validation_metric: str = LOSS
+    # "bucketing_field": None, # TODO: What is this?
+    learning_rate_warmup_epochs: float = schema.FloatRange(default=1.0, min=0.0)
+
+    class Meta:
+        unknown = INCLUDE
+
+
 class Trainer(BaseTrainer):
     """Trainer is a class that trains a model."""
 
+    @staticmethod
+    def get_schema_cls():
+        return TrainerConfig
+
     def __init__(
         self,
         model: ECD,
-        optimizer=None,
-        epochs=100,
-        regularization_lambda=0.0,
-        regularization_type=None,
-        learning_rate=0.001,
-        decay=False,
-        decay_rate=0.96,
-        decay_steps=10000,
-        staircase=False,
-        batch_size=128,
-        eval_batch_size=None,
+        # optimizer=None,
+        # epochs=100,
+        # regularization_lambda=0.0,
+        # regularization_type=None,
```

Comment:
Why comment these parameters out? Can we remove them?

Reply:
Just keeping them around for reference, to be taken out in final refactor :P